docs/guides/unity-catalog.md

```diff
@@ -0,0 +1,530 @@
+  host     = var.databricks_workspace_host
+  username = var.databricks_account_username
+  password = var.databricks_account_password
+}
+```
+
+Define the required variables
+
+```hcl
+variable "databricks_account_username" {}
+variable "databricks_account_password" {}
+variable "databricks_account_id" {}
+variable "databricks_workspace_host" {}
+
+variable "tags" {
+  default = {}
+}
+
+variable "region" {
+  default = "eu-west-1"
+}
+
+variable "databricks_workspace_ids" {
+  description = <<EOT
+  List of Databricks workspace ids to be enabled with Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g. ["111111111", "222222222"]
+  EOT
+  type        = list(string)
+}
+
+variable "databricks_users" {
+  description = <<EOT
+  List of Databricks users to be added at account-level for Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g ["first.last@domain.com", "second.last@domain.com"]
+  EOT
+  type        = list(string)
+}
+
+variable "databricks_unity_admins" {
+  description = <<EOT
+  List of Admins to be added at account-level for Unity Catalog.
+  Enter with square brackets and double quotes
+  e.g ["first.admin@domain.com", "second.admin@domain.com"]
+  EOT
+  type        = list(string)
+}
+
+variable "unity_admin_group" {
+  description = "Name of the admin group. This group will be set as the owner of the Unity Catalog metastore"
+  type        = string
+}
+//generate a random string as the prefix for AWS resources, to ensure uniqueness
+resource "random_string" "naming" {
+  special = false
+  upper   = false
+  length  = 6
+}
+
+locals {
+  prefix = "demo${random_string.naming.result}"
+}
+```
+
+## Configure AWS objects
+The first step is to create the required AWS objects:
+- An S3 bucket, which be the default storage location for managed tables in Unity Catalog. Please use a dedicated bucket for each metastore.
+- An IAM policy that provides Unity Catalog permissions to access and manage data in the bucket
+- An IAM role that will be assumed by Unity Catalog
+
+```hcl
+resource "aws_s3_bucket" "metastore" {
+  bucket = "${local.prefix}-metastore"
+  acl    = "private"
+  versioning {
+    enabled = false
+  }
+  force_destroy = true
+  tags = merge(local.tags, {
+    Name = "${local.prefix}-metastore"
+  })
+}
+
+resource "aws_s3_bucket_public_access_block" "root_storage_bucket" {
+  bucket             = aws_s3_bucket.metastore.id
+  ignore_public_acls = true
+  depends_on         = [aws_s3_bucket.metastore]
+}
+
+data "aws_iam_policy_document" "passrole_for_uc" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
+      identifiers = ["arn:aws:iam::414351767826:role/unity-catalog-prod-UCMasterRole-14S5ZJVKOTYTL"]
+      type        = "AWS"
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "sts:ExternalId"
+      values   = [var.databricks_account_id]
+    }
+  }
+}
+
+resource "aws_iam_policy" "unity_metastore" {
```

Comment:
Is this creating a new IAM policy? I thought that the customer should be creating the policy on their own -- they might need to edit the actions, unless the actions below are all required for UC.

Reply:
well, it's allowing UC to do all required actions with S3